src/server/engine_shard.cc

```diff
@@ -262,7 +262,7 @@ uint64_t TEST_current_time_ms = 0;
 
 ShardId Shard(string_view v, ShardId shard_num) {
   if (IsClusterShardedByTag()) {
-    v = LockTagOptions::instance().Tag(v);
+    return KeySlot(v) % shard_num;
```

Comment:
1. This change causes the tests to fail (tried it out locally). 

2. My understanding here is that you want to map multiple `prefixes` to the same `suffix` and with the `crc() % shard_num` will eventually end up in the same shard/slot. Is that right ? (from your example mget {foo}aaa {bar}aaa)

3. This change is not limited only in the scope of cluster because:
```
 89 bool IsClusterShardedByTag() {
 90   return IsClusterEnabledOrEmulated() || LockTagOptions::instance().enabled;
 91 } 
```

That means when `lock_on_hashtags` is enabled (even without cluster), we now hash differently.